### ISSUE TITLE
Single command for dev setup

### ISSUE THREAD

**Issue by akirabaruah** [cardinalitypuzzles/cardboard#249]
There are several steps/commands that folks must use to initially setup their local repo for development. We should simplify DX by providing a single command that "just works".

Some ideas from discussion with @cs-miller @pcvera:

- wrap everything into a `yarn` or `npm` script command like `yarn setup`
- run everything inside a docker container to ensure consistent environment
- idempotent: allow users to run the command several times without any issues. in other words, the command shouldn't assume any special "state"

**Comment by rgossiaux** [comment]
I think the idea was that you can just run `docker-compose up` (maybe with --build) after downloading the repo and it will get everything up and running.

**Comment by akirabaruah** [comment]
Yeah `docker-compose` currently handles running the "product" itself, but doesn't handle any dev tool setup like `docker` or git pre-commit hook installation. I think this issue captures the latter request for a simple "bootstrapping" shell script. Otherwise, running `docker-compose` assumes that the user has all of the prereq dependencies installed, with which some of our devs have faced some friction.

**Comment by npinsker** [comment]
closing in favor of the newer #754
